### PR TITLE
gh-111062: Separate linux build into a reusable workflow

### PR DESCRIPTION
**Pull request by NCLI**
* Issue: gh-111062

### CODE REVIEW

File: .github/workflows/reusable-build-ubuntu.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+name: Tests
```
Comment (dimaqq) [inline .github/workflows/reusable-build-ubuntu.yml]: @corona10 says remove this name

File: .github/workflows/reusable-build-ubuntu.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        type: string
+      ubuntu_version:
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  name: 'Ubuntu reusable'
```
Comment (dimaqq) [inline .github/workflows/reusable-build-ubuntu.yml]: @corona10 says remove this name

**Review by dimaqq — APPROVED** [review]
Looks fine 😄

**Comment by hugovk** [comment]
Looking good, let's first decide on the name, re: https://github.com/python/cpython/issues/111062#issuecomment-1784151515

`reusable-build-ubuntu.yml`, or `reusable-ubuntu.yml` to match `reusable-docs.yml`?

**Comment by corona10** [comment]
Duplicated with https://github.com/python/cpython/pull/111452

**Pull request closed without merging**